logging/testing/util.go

```diff
@@ -27,39 +26,14 @@ import (
 	"knative.dev/pkg/logging"
 )
 
-var (
-	loggers = make(map[string]*zap.SugaredLogger)
-	m       sync.Mutex
-)
-
 // TestLogger gets a logger to use in unit and end to end tests
 func TestLogger(t *testing.T) *zap.SugaredLogger {
-	m.Lock()
-	defer m.Unlock()
-
-	logger, ok := loggers[t.Name()]
-
-	if ok {
-		return logger
-	}
-
 	opts := zaptest.WrapOptions(
 		zap.AddCaller(),
 		zap.Development(),
 	)
 
-	logger = zaptest.NewLogger(t, opts).Sugar().Named(t.Name())
-	loggers[t.Name()] = logger
-
-	return logger
-}
-
-// ClearAll removes all the testing loggers.
-// `go test -count=X` executes runs in the same process, thus the map
-// persists between the runs, but the `t` will no longer be valid and will
-// cause a panic deep inside testing code.
-func ClearAll() {
```

Comment:
Perhaps we should leave this for now, since it's breaking.  Once the no-caching change lands we can remove all of the ClearAll() calls and safely remove this.

Reply:
@mattmoor Sounds good, I can leave the `ClearAll()` for now. What about `SetupFakeContextWithCancel`? Currently it creates a new logger every time it's called, so I think we need to update it to not create a new logger

Reply:
Sorry, what I meant was: if we leave an *empty* `ClearAll()` then this stages more cleanly.  Once the downstream repos import this (automatically, unless you are quick) we can just remove all the ClearAll calls, and then come back and clean up this empty stub.  Sorry for not being more clear before.

Reply:
I see. I've updated this PR with the change. That also means for a short while (before changes to serving and eventing lands to remove multiple calls to create loggers in the same test), we could have multiple loggers for the same test?